addons/website/static/src/components/configurator/configurator.js

```diff
@@ -582,11 +582,16 @@ async function getInitialState(services) {
         industries: results.industries,
         logo: results.logo ? 'data:image/png;base64,' + results.logo : false,
     };
-    r.industries = r.industries.map((industry, index) => ({
-        ...industry,
-        wordCount: industry.label.split(" ").length,
-        hitCountOrder: index,
-    }));
+    r.industries = r.industries.map((industry, index) => {
+        // Keep original label for display.
+        industry.lowerCaseLabel = industry.label.toLowerCase();
+        industry.synonyms = (industry.synonyms || "").toLowerCase();
+        return {
+            ...industry,
+            wordCount: industry.label.split(" ").length,
+            hitCountOrder: index,
+        };
+    });
```

Comment:
Can't this whole PR just become a one line diff by just adding a `toLowerCase()` in the other changed lines?
```js
if (terms.every(term => candidate.toLowerCase().includes(term))) {
    return true;
}
```

I guess you prefered to do the lower casing only once at startup instead of one time every search, but that seems useless and makes things more complicate that it could (should?) be.

I stress tested both changes and it actually doesn't change anything at all tbh, as the processing time of the filter+split+for+includes is taking 99% of the time compare to the lower casing if you add it.
See:
```js
console.time('test');
let cpt = 0;
for (let i = 0; i < 100; i++) {
  this.state.industries.filter((val, index) => {
            // To match, every term should be contained in either the label or a
            // synonym
            for (const candidate of [val.label, ...(val.synonyms || '').split(/[|,\n]+/)]) {
                cpt++;
                if (terms.every(term => candidate.toLowerCase().includes(term))) {
                    return true;
                }
            }
        });
}
console.log(cpt);
console.timeEnd('test');
```

I have the same values with or without the lower casing in the loop, ~1850 ms for a 788900 `cpt` "loop".

I'd just change the whole PR and go for the one liner, unless I missed something.

Reply:
That's what I initially did, but indeed doing it at load time brings nothing.